google/resource_compute_address.go

```diff
@@ -1,83 +1,108 @@
+//
+// ----------------------------------------------------------------------------
+//
+//     This file is automatically generated by Magic Modules and manual
+//     changes will be clobbered when the file is regenerated.
+//
+//     Please read more about how to change this file in
+//     .github/CONTRIBUTING.md.
+//
+// ----------------------------------------------------------------------------
+
 package google
 
 import (
 	"fmt"
+	"log"
+	"time"
 
 	"github.com/hashicorp/terraform/helper/schema"
 	"github.com/hashicorp/terraform/helper/validation"
-	computeBeta "google.golang.org/api/compute/v0.beta"
-)
-
-const (
-	addressTypeExternal = "EXTERNAL"
-	addressTypeInternal = "INTERNAL"
+	compute "google.golang.org/api/compute/v1"
 )
 
 func resourceComputeAddress() *schema.Resource {
 	return &schema.Resource{
 		Create: resourceComputeAddressCreate,
 		Read:   resourceComputeAddressRead,
 		Delete: resourceComputeAddressDelete,
+
 		Importer: &schema.ResourceImporter{
-			State: resourceComputeAddressImportState,
+			State: resourceComputeAddressImport,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(240 * time.Second),
+			Delete: schema.DefaultTimeout(240 * time.Second),
 		},
 
 		Schema: map[string]*schema.Schema{
-			"name": &schema.Schema{
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateRegexp(`^(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)$`),
 			},
-
-			"address_type": &schema.Schema{
+			"address": {
 				Type:     schema.TypeString,
+				Computed: true,
 				Optional: true,
 				ForceNew: true,
-				Default:  addressTypeExternal,
-				ValidateFunc: validation.StringInSlice(
-					[]string{addressTypeInternal, addressTypeExternal}, false),
 			},
-
-			"subnetwork": &schema.Schema{
-				Type:             schema.TypeString,
-				Optional:         true,
-				ForceNew:         true,
-				Computed:         true,
-				DiffSuppressFunc: linkDiffSuppress,
+			"address_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"INTERNAL", "EXTERNAL", ""}, false),
+				Default:      "EXTERNAL",
 			},
-
-			// address will be computed unless it is specified explicitly.
-			// address may only be specified for the INTERNAL address_type.
-			"address": &schema.Schema{
+			"description": {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Computed: true,
 			},
-
-			"network_tier": &schema.Schema{
+			"network_tier": {
 				Type:         schema.TypeString,
+				Computed:     true,
 				Optional:     true,
 				ForceNew:     true,
-				Computed:     true,
-				ValidateFunc: validation.StringInSlice([]string{"PREMIUM", "STANDARD"}, false),
+				ValidateFunc: validation.StringInSlice([]string{"PREMIUM", "STANDARD", ""}, false),
```

Comment:
Question: is the `""` redundant because of `Optional: true`?

Reply:
Nope- it solves a fairly specific use case where people want to be able to conditionally set this field. #800 was the issue report that caused us to start adding it for generated code.

Reply:
Oh right, I remember now.  Thanks!